### PR TITLE
Feat 4 facebook compatible

### PR DESCRIPTION
**Pull request by NgGeorge**
Completed the filter implementation to also work with Facebook. Additionally, the filters were optimized and cleaned to better follow style conventions.

### CODE REVIEW

File: cloaked.js [hunk 1]
```diff
@@ -1,93 +1,122 @@
...
+		});
+		searchString = searchString.substring(0, searchString.length - 2);
+
+		// Blur the spoiler's parent (excluding body+head) because it's important to block the spoiler's surrounding terms.
+		$(searchString).parent(":not('body')", ":not('head')").css('-webkit-filter', 'blur(5px)').click(function() {
+			$(this).css('-webkit-filter', '');
+		});
+	}
+
+	// searchForSpoilers and blockFacebookSpoilers still need to be refactored into one function
```
Comment (k-chung) [inline cloaked.js:77]: Add the word TODO to make this clearer

File: cloaked.js [hunk 2]
```diff
@@ -1,93 +1,122 @@
...
+	}
+
+	// Case insensitive jquery contains
+	jQuery.expr[':'].icontains = function(element, index, match) {
+		return jQuery(element).text().toUpperCase()
+		.indexOf(match[3].toUpperCase()) >= 0;
+	};
+
+	// Run script
+	$(function () {
```
Comment (k-chung) [inline cloaked.js:109]: Is this a separate thing from the `(function ($) {` near the top of the page?

Comment (NgGeorge) [reply cloaked.js:109]: Yes, it's another self executing function

File: cloaked.js (hunk 1 above)
Comment (NgGeorge) [reply cloaked.js:77]: I decided against doing it, I ran into some issues that I wasn't able to figure out.

**Pull request merged**